tasks/size_contracts.js

```diff
@@ -41,37 +42,45 @@ task(
   );
 
   const previousSizes = {};
+  const previousInitSizes = {};
 
   if (fs.existsSync(outputPath)) {
     const previousOutput = await fs.promises.readFile(outputPath);
 
     JSON.parse(previousOutput).forEach(function (el) {
-      previousSizes[el.fullName] = el.size;
+      previousSizes[el.fullName] = el.deploySize;
+      previousInitSizes[el.fullName] = el.initSize;
     });
   }
 
   await Promise.all(fullNames.map(async function (fullName) {
     if (config.only.length && !config.only.some(m => fullName.match(m))) return;
     if (config.except.length && config.except.some(m => fullName.match(m))) return;
 
-    const { deployedBytecode } = await hre.artifacts.readArtifact(fullName);
-    const size = Buffer.from(
+    const { deployedBytecode, bytecode } = await hre.artifacts.readArtifact(fullName);
+    const deploySize = Buffer.from(
       deployedBytecode.replace(/__\$\w*\$__/g, '0'.repeat(40)).slice(2),
       'hex'
     ).length;
+    const initSize = Buffer.from(
+      bytecode.replace(/__\$\w*\$__/g, '0'.repeat(40)).slice(2),
+      'hex'
+    ).length;
 
     outputData.push({
       fullName,
       displayName: config.disambiguatePaths ? fullName : fullName.split(':').pop(),
-      size,
-      previousSize: previousSizes[fullName] || null,
+      deploySize,
+      previousDeploySize: previousSizes[fullName] || null,
+      initSize,
+      previousInitSize: previousInitSizes[fullName] || null,
     });
   }));
 
   if (config.alphaSort) {
     outputData.sort((a, b) => a.displayName.toUpperCase() > b.displayName.toUpperCase() ? 1 : -1);
   } else {
-    outputData.sort((a, b) => a.size - b.size);
+    outputData.sort((a, b) => a.deploySize - b.deploySize);
```

Comment:
It's unfortunate that sorting by deploy size will not guarantee that contracts with excessive init size will be together in the output table.

Reply:
Another reason to maybe split this into two tasks in a future release.